src/composite/tag/tag.jl

```diff
@@ -8,11 +8,8 @@ but tag the block with some extra information.
 """
 abstract type TagBlock{BT, N, T} <: AbstractContainer{BT, N, T} end
 
+# forward content properties
 cache_key(tb::TagBlock) = cache_key(content(tb))
 occupied_locs(x::TagBlock) = occupied_locs(content(x))
 
 Base.:(==)(a::TB, b::TB) where {TB<:TagBlock} = content(a) == content(b)
```

Comment:
Have you tested for `ScaleBlock`? BTW, should scale be a tagblock? tagblock should not allow free parameters by itself, right? Maybe we should handle this problem later.

Reply:
Yes, I think would like to move `Scale` to be under `AbstractBlock`.

Reply:
Container is better, I think.